index.js

```diff
@@ -94,17 +84,65 @@ function RedisClient(stream, options) {
     this.parser_module = null;
     this.selected_db = null; // Save the selected db here, used when reconnecting
     this.old_state = null;
-    this.options = options;
 
     this.install_stream_listeners();
     events.EventEmitter.call(this);
 }
 util.inherits(RedisClient, events.EventEmitter);
 
+RedisClient.prototype.create_stream = function() {
+    var options = this.options;
+    var connection_options = this.connection_options;
+    var tls_options = options.tls;
+    var stream;
+
+    if (!connection_options) {
+        connection_options = this.connection_options = {};
+        if (options.path) {
+            //unix
+            this.address = connection_options.path = options.path;
+        } else {
+            //tcp or tls
+            connection_options.port = options.port || default_port;
+            connection_options.host = options.host || default_host;
+            connection_options.family = (options.family === 'IPv6') ? 6 : 4;
+            if (tls_options) {
+                Object.keys(tls_options).forEach(function(opt) {
+                    connection_options[opt] = tls_options[opt];
+                });
+            }
+            this.address = connection_options.host + ':' + connection_options.port;
+        }
+    }
+
+    if (tls_options) {
+	    stream = tls.connect(connection_options);
+    } else {
+	    stream = net.createConnection(connection_options);
+    }
+
+    var self = this;
+    if (!stream.cork) {
+        this.pipeline = 0;
+        this.cork = noop;
+        this.once('ready', function () {
+            self.cork = function (len) {
+                self.pipeline = len;
+                self.pipeline_queue = new Queue(len);
+            };
+        });
+        stream.uncork = noop;
+        this.write = this.writeStream;
+    }
```

Comment:
Good catch. This is not necessary after #901 has landed though. I fixed the logic so it's independent from the used stream.